### PR TITLE
Modernize moffle dependencies

### PR DESCRIPTION
**Pull request by nolanlum**
- Lock dependencies via `pip-tools`
- Update form validation for new wtf api
- Update for new flask-babel api
- Switch to authlib OAuth (flask-oauthlib is unmaintained)
- Replace use of deprecated jinja escaping functionality
- Update to new cachetools library and API
- Update usages of werkzeug API

Temporarily disables monkeypatching perf improvements mainly because I think I was too lazy to fix them,,,

### CODE REVIEW

File: requirements.in [hunk 1]
```diff
@@ -0,0 +1,14 @@
+Authlib
+cachetools
+elasticsearch
+elasticsearch-dsl
```
Comment (nattofriends) [inline requirements.in:4]: think these never quite got used

**Review by nattofriends — APPROVED** [review]
probably fine to give up on the monkey patches, they were not meant to last 10 years anyway

File: requirements.in (hunk 1 above)
Comment (nolanlum) [reply requirements.in:4]: yeah... `grep.py` in master actually already imports from elasticsearch and elasticsearch_dsl so figured i'd add them here even if they're not used, if only to make the import work.

**Comment by nolanlum** [comment]
don't have push so you can merge whenever

**Pull request merged**